File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        Set<String> usedWords = new HashSet<>();
        usedWords.add(words[0]);

        for (int i = 1; i < words.length; i++) {
            String prevWord = words[i - 1];
            String currWord = words[i];

            if (prevWord.charAt(prevWord.length() - 1) != currWord.charAt(0) || 
                usedWords.contains(currWord)) {
                return new int[]{(i % n) + 1, (i / n) + 1};
            }
            usedWords.add(currWord);
        }
        return new int[]{0, 0};
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    used_words = set()
    used_words.add(words[0])

    for i in range(1, len(words)):
        word = words[i]
        prev_word = words[i - 1]

        if word[0] != prev_word[-1] or word in used_words:
            person_num = (i % n) + 1
            turn_num = (i // n) + 1
            return [person_num, turn_num]

        used_words.add(word)

    return [0, 0]